react/src/components/ResourceAllocationFormItems.tsx

```diff
@@ -1206,7 +1208,7 @@ const ResourceAllocationFormItems: React.FC<
         </Card>
       ) : null}
       {/* TODO: Support cluster mode */}
-      {!baiClient._config.hideAgents && (
+      {!baiClient._config.hideAgents && enableAgentSelect && (
```

Comment:
Let's move this condition to the `SessionLauncherPage` that uses this component.
```
<ResourceAllocationFormItems 
   enableAgentSelect={baiClient._config.hideAgents}/>
```
This change would simplify the ResourceAllocationFormItems component, making it more focused on state management, while also clarifying the control logic in the parent component.